rdr_service/api/mayolink_api.py

```diff
@@ -32,9 +77,9 @@ def _get_credentials(cls, credentials_key):
 
         return credentials.get('username'), credentials.get('password'), credentials.get('account')
 
-    @auth_required(RDR_AND_PTC)
-    def post(self, order):
-        xml = self.__dict_to_mayo_xml__(order)
+    def post(self, order: MayoLinkOrder):
+        check_auth(RDR_AND_PTC)
```

Comment:
Is there a reason to move away from the decorator?

Reply:
The decorator complicates things in a few ways when the only purpose seems to be to make sure we're only getting RDR and PTC users in this code. It was frustrating getting a test around it here since the `auth_required` method would require the use of the database for saving a request log. But I think that also was causing some strangeness in the request log table when something went wrong with the requests: we would see two records being logged for the same request. I think this fixes it, but even if it doesn't it definitely simplifies the test.